src/core/qgsmapsettingsutils.cpp

```diff
@@ -17,9 +17,49 @@
 
 #include "qgsmapsettings.h"
 #include "qgsmapsettingsutils.h"
+#include "qgspallabeling.h"
+#include "qgstextrenderer.h"
+#include "qgsvectorlayer.h"
 
 #include <QString>
 
+bool QgsMapSettingsUtils::containsAdvancedEffects( const QgsMapSettings &mapSettings )
```

Comment:
Was this copied from composer? Can we call it elsewhere and remove the duplicate code, m